packaging/docker-image/Makefile

```diff
@@ -0,0 +1,32 @@
+GENERIC_UNIX_ARCHIVE ?= $(wildcard $(PACKAGES_DIR)/rabbitmq-server-generic-unix-*.tar.xz)
+
+ifneq ($(filter-out clean,$(MAKECMDGOALS)),)
+ifeq ($(GENERIC_UNIX_ARCHIVE),)
+$(error Cannot find generic-unix archive; please specify GENERIC_UNIX_ARCHIVE)
+endif
+
+VERSION ?= $(patsubst rabbitmq-server-generic-unix-%.tar.xz,%,$(notdir $(GENERIC_UNIX_ARCHIVE)))
+ifeq ($(VERSION),)
+$(error Cannot determine version; please specify VERSION)
+endif
+endif
+
+OTP_VERSION ?= 23.2.4
+OTP_SHA256 ?= e72aa084907e0f34f932cf00caa33aba93147b0a7c9c35569d6bd1c402f532de
+
+all: dist
+
+dist:
+	xzcat $(GENERIC_UNIX_ARCHIVE) | tar xvf -
+	docker build --pull \
+	  --build-arg PGP_KEYSERVER=pgpkeys.uk \
+	  --build-arg OTP_VERSION=$(OTP_VERSION) \
+	  --build-arg OTP_SHA256=$(OTP_SHA256) \
+	  --build-arg RABBITMQ_VERSION=$(VERSION) \
+	  --build-arg RABBITMQ_BUILD=rabbitmq_server-$(VERSION) \
+	  --tag pivotalrabbitmq/rabbitmq:$(VERSION) \
+	  .
+	docker push pivotalrabbitmq/rabbitmq:$(VERSION)
```

Comment:
I think we should remove:
```
docker push pivotalrabbitmq/rabbitmq:$(VERSION)
```
Build the docker image based on source code is useful for k8s tests. 
We'd like to avoid to push the tag each time.
We can:
 - just remove the `push` and leave to the user the decision
 - create anther make step like `make push-docker-image`


@gerhard thought?

Reply:
Yes, pushing should be a separate target. We do the same in rabbitmq_prometheus: https://github.com/rabbitmq/rabbitmq-server/blob/0dc46b41de9f5f51dfa12bfb5e4336c7b32e2e6d/deps/rabbitmq_prometheus/Makefile#L130-L133

Reply:
I'd vote for the separate target

Reply:
OK